src/Installer/dnup/ArchiveInstallationValidator.cs

```diff
@@ -1,17 +1,145 @@
+        {
+            string runtimeDirectory = Path.Combine(installRoot, "shared", runtimeMoniker, resolvedVersion);
+            return Directory.Exists(runtimeDirectory);
+        }
+
+        return false;
+    }
+
+    private bool ValidateWithHostFxr(string installRoot, ReleaseVersion resolvedVersion, InstallComponent component)
+    {
+        try
+        {
+            ConfigureHostFxrResolution(installRoot);
+
+            var bundleProvider = new NETBundlesNativeWrapper();
+            NetEnvironmentInfo environmentInfo = bundleProvider.GetDotnetEnvironmentInfo(installRoot);
+
+            if (component == InstallComponent.SDK)
+            {
+                string expectedPath = Path.Combine(installRoot, "sdk", resolvedVersion.ToString());
+                return environmentInfo.SdkInfo.Any(sdk =>
+                    string.Equals(sdk.Version.ToString(), resolvedVersion.ToString(), StringComparison.OrdinalIgnoreCase) &&
+                    DnupUtilities.PathsEqual(sdk.Path, expectedPath));
+            }
+
+            if (!RuntimeMonikerByComponent.TryGetValue(component, out string? runtimeMoniker))
+            {
+                return false;
+            }
+
+            string expectedRuntimePath = Path.Combine(installRoot, "shared", runtimeMoniker, resolvedVersion.ToString());
+            return environmentInfo.RuntimeInfo.Any(runtime =>
+                string.Equals(runtime.Name, runtimeMoniker, StringComparison.OrdinalIgnoreCase) &&
+                string.Equals(runtime.Version.ToString(), resolvedVersion.ToString(), StringComparison.OrdinalIgnoreCase) &&
+                DnupUtilities.PathsEqual(runtime.Path, expectedRuntimePath));
+        }
+        catch
+        {
+            return false;
+        }
+    }
+
+    private static void ConfigureHostFxrResolution(string installRoot)
+    {
+        if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            return;
+        }
+
+        if (AppContext.GetData(HostFxrRuntimeProperty) is not null)
+        {
+            return;
+        }
+
+        string? hostFxrPath = FindHostFxrLibrary(installRoot);
+        if (hostFxrPath is not null)
+        {
+            AppContext.SetData(HostFxrRuntimeProperty, hostFxrPath);
+        }
+    }
+
+    private static string? FindHostFxrLibrary(string installRoot)
+    {
+        string hostFxrDirectory = Path.Combine(installRoot, "host", "fxr");
+        if (!Directory.Exists(hostFxrDirectory))
+        {
+            return null;
+        }
+
+        string libraryName = RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
+            ? "hostfxr.dll"
+            : RuntimeInformation.IsOSPlatform(OSPlatform.OSX)
+                ? "libhostfxr.dylib"
+                : "libhostfxr.so";
+
+        return Directory.EnumerateFiles(hostFxrDirectory, libraryName, SearchOption.AllDirectories)
+            .OrderByDescending(File.GetLastWriteTimeUtc)
```

Comment:
I don't think this is the best approach at determining which hostfxr to use, though the cases where this would be called is rare. I would be ok with removing this or instead searching the latest directory